File: xssfinder-test/src/test/java/org/xssfinder/testsite/simple/page/HomePage.java
package org.xssfinder.testsite.simple.page;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.xssfinder.CrawlStartPoint;
import org.xssfinder.Page;
import org.xssfinder.SubmitAction;

@Page
public class HomePage {
    private final WebDriver driver;

    public HomePage(WebDriver driver) {
        this.driver = driver;
    }

    @SubmitAction
    public HomePage unsafeSubmit() {
        driver.findElement(By.id("unsafeSubmit")).click();
        return this;
    }

    @SubmitAction
    public HomePage safeSubmit() {
        driver.findElement(By.id("safeSubmit")).click();
        return this;
    }

    public LoginPage logout() {
        driver.findElement(By.linkText("log out")).click();
        return new LoginPage(driver);
    }
}

File: xssfinder-php-executor/test/XssFinder/TestSite/Page/HomePage.php
<?php

namespace XssFinder\TestSite\Page;

/**
 * @page
 */
class HomePage
{
    /** @var \WebDriver */
    private $_driver;

    public function __construct(\WebDriver $driver)
    {
        $this->_driver = $driver;
    }

    /**
     * @submitAction
     * @return HomePage
     */
    public function unsafeSubmit()
    {
        $this->_driver->findElement(\WebDriverBy::id('unsafeSubmit'))->click();
        return $this;
    }

    /**
     * @submitAction
     * @return HomePage
     */
    public function safeSubmit()
    {
        $this->_driver->findElement(\WebDriverBy::id('safeSubmit'))->click();
        return $this;
    }

    /**
     * @return LoginPage
     */
    public function logout()
    {
        $this->_driver->findElement(\WebDriverBy::linkText('log out'))->click();
        return $this;
    }

    /**
     * @return UnreachedPage
     */
    public function throwException()
    {
        throw new \Exception('Intentional thrown exception');
    }
}